freqtrade/data/history/history_utils.py

```diff
@@ -152,71 +152,90 @@ def _load_cached_data_for_updating(pair: str, timeframe: str, timerange: Optiona
     return data, start_ms
 
 
-def _download_pair_history(datadir: Path,
-                           exchange: Exchange,
-                           pair: str, *,
+def _download_pair_history(datadir: Path, exchange: Exchange, pair: str, *,
                            timeframe: str = '5m',
                            timerange: Optional[TimeRange] = None,
                            data_handler: IDataHandler = None) -> bool:
-    """
-    Download latest candles from the exchange for the pair and timeframe passed in parameters
-    The data is downloaded starting from the last correct data that
-    exists in a cache. If timerange starts earlier than the data in the cache,
-    the full data will be redownloaded
-
-    Based on @Rybolov work: https://github.com/rybolov/freqtrade-data
 
-    :param pair: pair to download
-    :param timeframe: Timeframe (e.g "5m")
-    :param timerange: range of time to download
-    :return: bool with success state
-    """
     data_handler = get_datahandler(datadir, data_handler=data_handler)
 
     try:
+        since, until = None, None
+        if timerange:
+            since, until = timerange.startts, timerange.stopts
+
         logger.info(
-            f'Download history data for pair: "{pair}", timeframe: {timeframe} '
-            f'and store in {datadir}.'
+            f'Downloading history data for pair: "{pair}", timeframe: '
+            f'{timeframe} and storing in "{datadir}"'
         )
 
-        # data, since_ms = _load_cached_data_for_updating_old(datadir, pair, timeframe, timerange)
-        data, since_ms = _load_cached_data_for_updating(pair, timeframe, timerange,
```

Comment:
I think the loading of existing data should still be left in this method - just that the method now should return both start and end dates (on top of the data, obviously).

That'll make it easier to test and make this method less ... complicated / long.